core/store/src/trie/state_parts.rs

```diff
@@ -267,11 +267,11 @@ impl Trie {
                 .iter()
                 .map(|entry| (*entry.hash(), entry.payload().to_vec().into())),
         );
-        let final_trie =
-            Trie::new(Arc::new(TrieMemoryPartialStorage::new(all_nodes)), self.root, None);
-
+        let partial_storage = Arc::new(TrieMemoryPartialStorage::new(all_nodes));
+        let final_trie = Trie::new(partial_storage.clone(), self.root, None);
```

Comment:
cloning the partial storage seems expensive - can we avoid it? The old semantics of getting the storage back from the trie seems fine.

Reply:
please ignore if everything is Arc-ed already

Reply:
Yeah its an Arc, I should've used `Arc::clone` anyhow, as is our style guide.